include/xnnpack.h

```diff
@@ -1828,6 +1828,20 @@ struct xnn_external_value {
   void* data;
 };
 
+/// Reshape an external value.
+///
+/// @param external_id - external ID for the Value. The ID must be within the range of reversed Value IDs specified on
+///                      the Subgraph creation. If the external ID is XNN_INVALID_VALUE_ID, an internal ID will be
+///                      created for the Value.
+/// @param num_dims - number of dimensions in the shape.
+/// @param dims - pointer to an array of @a num_dims shape dimensions. If num_dims is 0, this pointer can be NULL.
+///               XNNPACK does not keep any pointers to this array after the function returns.
```

Comment:
What will the API look like to get back the final output shape? Since this ptr is not something we plan to hold or update.

Reply:
See https://github.com/google/XNNPACK/pull/5832 for this API